src/Moryx/Serialization/DefaultSerialization.cs

```diff
@@ -34,7 +34,12 @@ public DefaultSerialization()
         /// <see cref="ICustomSerialization"/>
         public virtual IEnumerable<PropertyInfo> GetProperties(Type sourceType)
         {
-            return sourceType.GetProperties(BindingFlags.Instance | BindingFlags.Public);
+            var properties = sourceType.GetProperties(BindingFlags.Instance | BindingFlags.Public);
+            var hasEntrySerializeAttribute = sourceType.GetCustomAttribute(typeof(EntrySerializeAttribute)) != null;
+            //ignore the entry serialize attribute on a class
+            return hasEntrySerializeAttribute ?
+                properties.Where(x => x.GetCustomAttributes(typeof(EntrySerializeAttribute)).Count() > 0) :
```

Comment:
I think this method get's called for all layers. We can not prohibit it in general, as this would break multiple usages down the tree.

Reply:
Also, this type of serialization shoudl not check for `EntrySerialize` at all. There is a dedicated implementation with a more sophistacated check.

Reply:
@Toxantron  Are you referring to the `ResourceSerialization` or what?